### PR TITLE
test: configure Solana gateway program id for E2E tests

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

- [x] Removed hardcoded Solana gateway program ID and use configured address in the `localnet.yml` file.

Closes: https://github.com/zeta-chain/node/issues/3148

# How Has This Been Tested?

- [x] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Release Notes

- **New Features**
  - Whitelist functionality for SPL tokens on Solana.
  - Integration of SPL deposits and withdrawals.
  - Enhanced end-to-end testing for Solana gateway configurations.

- **Bug Fixes**
  - Resolved issues with out-of-gas errors during omnichain calls.
  - Fixed configuration resolution problems in the orchestrator.

- **Refactor**
  - Improved the zetaclient and CLI for better performance and usability.
  - Updated contract address handling for consistency across configurations.

These updates significantly enhance the functionality and reliability of the ZetaChain ecosystem.

### CODE REVIEW

File: zetaclient/testutils/constant.go [hunk 1]
```diff
@@ -39,12 +39,14 @@ const (
 	EventERC20Withdraw = "Withdrawn"
 )
 
+// OldSolanaGatewayAddressDevnet is the old gateway address deployed on Solana devnet
+const OldSolanaGatewayAddressDevnet = "94U5AHQMKkV5txNJ17QPXWoh474PheGou6cNP2FEuL1d"
```
[the remark this answers is not in the document]
Comment (ws4charlie) [reply zetaclient/testutils/constant.go:43]: Both old program and the new program `ZETAjseVjuFsxdRxo6MmTCvqFwb3ZHUx56Co3vCmGis` have SAME `Devnet` chainID and can't be put in same map. The former was need by some unit tests.

File: cmd/zetae2e/config/local.yml [hunk 2]
```diff
@@ -116,4 +116,7 @@ contracts:
     custody: "0xff3135df4F2775f4091b81f4c7B6359CfA07862a"
     erc20: "0xbD1e64A22B9F92D9Ce81aA9B4b0fFacd80215564"
     test_dapp: "0xBFF76e77D56B3C1202107f059425D56f0AEF87Ed"
-    gateway: "0xF0deebCB0E9C829519C4baa794c5445171973826"
\ No newline at end of file
+    gateway: "0xF0deebCB0E9C829519C4baa794c5445171973826"
+  solana:
+    gateway_program_id: "94U5AHQMKkV5txNJ17QPXWoh474PheGou6cNP2FEuL1d"
+    spl: ""
```
[the remark this answers is not in the document]
Comment (ws4charlie) [reply cmd/zetae2e/config/local.yml:122]: Local E2E tests doesn't need it. I put it here as a placeholder so we can copy and paste the section for live test. But I can remove it as well.

[the remark this answers is not in the document]
Comment (gartnera) [reply cmd/zetae2e/config/local.yml:122]: It does if you want to run `zetae2e` on your host right?

File: zetaclient/testutils/constant.go [hunk 3]
```diff
@@ -39,12 +39,14 @@ const (
 	EventERC20Withdraw = "Withdrawn"
 )
 
+// OldSolanaGatewayAddressDevnet is the old gateway address deployed on Solana devnet
```
Comment (lumtis) [inline zetaclient/testutils/constant.go:42]: Can we explain why do we need it?

File: cmd/zetae2e/config/localnet.yml [hunk 4]
```diff
@@ -98,4 +98,8 @@ rpcs:
   ton_sidecar_url: "http://ton:8000"
   zetacore_grpc: "zetacore0:9090"
   zetacore_rpc: "http://zetacore0:26657"
-# contracts will be populated on first run
+contracts:
+# configure localnet solana gateway program id
+  solana:
+    gateway_program_id: "94U5AHQMKkV5txNJ17QPXWoh474PheGou6cNP2FEuL1d"
+    spl: ""
```
[the remark this answers is not in the document]
Comment (ws4charlie) [reply cmd/zetae2e/config/localnet.yml:105]: @gartnera Can you clarify? My understanding was. that the Solana program is a predetermined address before contract deployment.

[the remark this answers is not in the document]
Comment (gartnera) [reply cmd/zetae2e/config/localnet.yml:105]: The address is dependent on the deployer key which I regenerate in the developnet deployment.

File: zetaclient/testutils/constant.go (hunk 3 above)
Comment (ws4charlie) [reply zetaclient/testutils/constant.go:42]: We have a few (four) early transactions used by unit tests. Those transactions were performed on the early program ID `94U5AHQMKkV5txNJ17QPXWoh474PheGou6cNP2FEuL1d` in Devnet, for example: 

`zetaclient/testdata/solana/chain_901_inbound_tx_result_MS3MPLN7hkbyCZFwKqXcg8fmEvQMD74fN6Ps2LSWXJoRxPW5ehaxBorK9q1JFVbqnAvu9jXm6ertj7kT7HpYw1j`
`zetaclient/testdata/solana/chain_901_outbound_tx_result_5iBYjBYCphzjHKfmPwddMWpV2RNssmzk9Z8NNmV9Rei71pZKBTEVdkmUeyXfn7eWbV8932uSsPfBxgA7UgERNTvq`

File: cmd/zetae2e/config/localnet.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ws4charlie) [reply cmd/zetae2e/config/localnet.yml:105]: Is there any way to get address when executing `solana program deploy gateway.so` in the `start-solana.sh`. Or is there any source to get the address?

File: cmd/zetae2e/config/local.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ws4charlie) [reply cmd/zetae2e/config/local.yml:122]: Yes, if I want to run a single e2e test on my local host.

File: cmd/zetae2e/config/localnet.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (gartnera) [reply cmd/zetae2e/config/localnet.yml:105]: Oh interesting I didn't realize we were doing it that way. In that case yeah we will need to hardcode this I think.

[the remark this answers is not in the document]
Comment (ws4charlie) [reply cmd/zetae2e/config/localnet.yml:105]: Cool.

**Review by lumtis — APPROVED** [review]

**Pull request merged**